internal/pkg/v2/infrastructure/redis/event.go

```diff
@@ -25,6 +26,9 @@ const (
 	EventsCollectionReadings   = EventsCollection + ":readings"
 )
 
+// deleteEventsByIdChannel channel used to delete events asynchronously
+var deleteEventsByIdChannel = make(chan []string, 50)
```

Comment:
This should be in Core Data application layer. App layer should request the ids from the DB layer and pass to the channel .

Reply:
Per comments from Cloud below, I sill leave the channel in the infrastructure layer.